src/rust/engine/BUILD

```diff
@@ -0,0 +1,36 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+files(
+    name="rust_sources",
+    sources=[
+        "Cargo.lock",
+        "build.rs",
+        "**/Cargo.toml",
+        "**/*.rs",
+        "**/*.proto",
+        ".cargo/config",
+        "rust-toolchain",
+    ],
+)
+
+shell_sources(name="shell")
+
+shell_command(
+    name="engine-and-client",
+    command=f"./cargo_build_shim.sh --features=extension-module -p engine -p client",
```

Comment:
Also, we could techincally split this into 2 `shell_command`s. One for `engine` and one for `client`, so we'd compile them in parallel.

A Rustacean could tell me if thats better, or worse (depending on if Rust is already compiling them in parallel behind the scenes)

Reply:
Cargo will compile in parallel. Having separate targets in Pants in parallel will likely be significantly slower when building both, because each sandbox will need to do a "full" recompile even of the shared dependencies. (E.g. both the client and the engine depend on tokio, and those will need to be compiled fresh, which is a bit of a pain.)